src/actions/breakpoints.js

```diff
@@ -82,6 +89,41 @@ export function enableBreakpoint(location: Location) {
   };
 }
 
+export function syncBreakpoint(location, referenceBreakpoint) {
+  return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
+    if (_breakpointExists(getState(), location)) {
+      return Promise.resolve();
+    }
+
+    const breakpoint = _createBreakpoint(location, referenceBreakpoint);
+
+    let promise = addClientBreakpoint(
```

Comment:
I'm a bit confused, so I'll walk through the bug and two workflows as I see them. I could be missing something, it could be what you're already doing too.  I'll outline a simple case and syncing case for both workflows. To be clear, we can do the simple case and fix the bug.

#### Bug

1. User is working on an app that uses source maps (i.e. namely bundle.js w/ file a.js, b.js, c.js)
2. User debugs the app and adds a bp to b.js line 12, with the generated location bundle.js#45
3. User adds code to a.js 
4. User reloads the app and see the debugger paused on b.js line 4, which happens to line up with the generated location bundle.js#45!

We will fix this, either simply moving the breakpoint to b.js line 4 when the debugger reloads or ideally re-removing the breakpoint at line bundle.js#45 and adding a breakpoint at the new location...

--- 

#### simple case

1. User adds a Breakpoint

  * addBreakpoint is called with an original location 
  * generated location is calculated based on the source's sourcemap
  * a pending breakpoint is added with the generated location

2. User reloads page / User re-opens debugger

  * breakpoints reducer adds pending breakpoints 
  * newSource is called and the debugger attempts to sync the breakpoint
       * checks if there is a pending breakpoint
       * uses the pending bps generated location to recalculate the original location
       * adds a new breakpoint to the store


---

#### syncing case

1. User adds a Breakpoint

  * addBreakpoint is called with an original location 
  * generated location is calculated based on the source's sourcemap
  * a pending breakpoint is added with the generated location

2. User reloads page / User re-opens debugger

  * breakpoints reducer adds pending breakpoints 
  * newSource is called and the debugger attempts to sync the breakpoint
       * checks if there is a pending breakpoint
       * uses the pending bps generated location to recalculate the original location
       * **if the new original location is different from the pending breakpoint original location**
          * removes the old breakpoint on the server
          * adds the new breakpoint to the server
       * adds the new breakpoint to the store